### DISCUSSION TITLE
current note context

### DISCUSSION THREAD

**Discussion by jhmonroe** [logancyang/obsidian-copilot#733] [General]
Hello! LOVE COPILOT!

I see in the latest version you got rid of the SEND NOTE (aka current note context) button.

From a UX standpoint, this means instead of a single click, now to provide this info we have to type out
[
[
search for note name
enter to select
then give instructions

It went from being 1 action to being 4

Why? It was so simple before.

Perhaps dropdown could have "Current Note" context option along with Vault QA? (i think maybe you had this a few versions ago?)

Thank you and appreciate you!!

**Comment by logancyang** [comment]
Hi @jhmonroe, thanks for the input! In the next few iterations the chat input will be revamped to the style of Cursor with a "context menu". You can add/remove notes in the context. For now, I'm gradually moving toward that pattern by removing this button and preferring explicit mentions. Since explicit mentions will also get appended to context menu on the fly.

(BTW, this post is better as a post in discussion)

**Reply by jhmonroe** [reply]
@logancyang sorry I'm used to WP forum where issues and feature requests both go in issues!

**Reply by enesbala5** [reply]
Has this been completed? I don't see any way to add context to the chat

**Reply by jhmonroe** [reply]
In design school a professor once said "don't throw the baby out with the bathwater" and I'll never forget it. What she meant was that every time we redesigned something we shouldn't get rid of something that is already working or that works well.

Apple, for instance, is notorious for this. They'll remove subtle quality-of-life things that worked well every time they redesign their UI. I've noticed the last few years that the trend towards minimalism and removing buttons means that everything becomes hidden or the truly easy things don't seem easy anymore. 

Because Obsidian is designed in such a user-friendly way, I worry that this tendency to move toward having to essentially "write code" to interact with a plugin (ie: explicit mentions every time you want to ask a question about the current note), or to have to use the command palette all the time makes plugins more complicated than they need to be. If plugin users were people who write code they would just write their own code. Plugin users, to my mind, are people who need GUI more than the need CLI. Or in the Obsidian ecosystem it's at least a 50/50 balance of super nerds and average folks who are more visual.

The context mode sounds like a good idea! And bringing back the thing it had that worked well.
Presumably it should be sticky and stick with the latest choice. The auto-appending is great too!
Current Note (for people who want to work in current note mode often)
Entire Vault
Etc
